### PR TITLE
vdirsyncer infrastructure 

### PR DESCRIPTION
**Pull request by ayyess**
I've added to your work on using vdirsyncer and calendar accounts to work with contact accounts using khal.
There is also a vdirsyncer service and I've added clientIdCommand and clientSecretCommand to correspond to passwordCommand.

### CODE REVIEW

File: modules/accounts/calendar.nix [hunk 1]
```diff
@@ -36,11 +45,32 @@ in
...
+        let
+          primaries =
+            catAttrs "name"
+            (filter (a: a.primary)
+            (attrValues cfg.accounts));
+        in
+          {
+            assertion = length primaries == 1;
+            message =
+              "Must have exactly one primary mail account but found "
```
Comment (loewenheim) [inline modules/accounts/calendar.nix]: I think this should be “contacts account”.

File: modules/programs/khal.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    home.packages =  [ pkgs.khal ];
+
+    xdg.configFile."khal/config".text = concatStringsSep "\n" (
+    [
+      "[calendars]"
+    ]
+    ++ (mapAttrsToList (name: value: concatStringsSep "\n"
+      ([
+        ''[[${name}]]''
+        ''path = ${value.vdirsyncer.local.path + (if value.khal.type == "discover" then "/*" else "/")}''
```
Comment (loewenheim) [inline modules/programs/khal.nix]: I think the fact that you need to access the `vdirsyncer` configuration here suggests that the local calendar configuration should be moved to the account directly, at least the parts that don’t immediately reference `vdirsyncer`.

**Review by loewenheim — APPROVED** [review]

File: modules/programs/khal.nix (hunk 2 above)
Comment (ayyess) [reply modules/programs/khal.nix]: Very true. I was trying to keep in line with what you had already done so I've moved it out now.

File: modules/programs/khal.nix [hunk 3]
```diff
@@ -37,7 +37,7 @@ in
     [
     (generators.toINI {} {
       default = {
-        default_calendar = head (mapAttrsToList (n: v: baseNameOf v.vdirsyncer.local.path) (filterAttrs (_: a: a.primary) khalAccounts));
+        default_calendar = head (mapAttrsToList (n: v: baseNameOf v.path) (filterAttrs (_: a: a.primary) khalAccounts));
```
Comment (loewenheim) [inline modules/programs/khal.nix]: This line fails with the somewhat cryptic “list index 0 is out of bounds” if `khal` isn’t enabled for any account.

File: modules/programs/vdirsyncer-accounts.nix [hunk 4]
```diff
@@ -9,7 +9,7 @@ let
   remoteModule = types.submodule {
     options = {
       type = mkOption {
-        type = types.enum [ "caldav" "http" "google_calendar" ];
+        type = types.enum [ "caldav" "http" "google_calendar" "google_contacts" ];
```
Comment (loewenheim) [inline modules/programs/vdirsyncer-accounts.nix:12]: There needs to be a `carddav` collection type for contacts.

**Comment by loewenheim** [comment]
Thank you very much for your contributions so far. This is the first pull request I’ve gotten :)

**Pull request merged**